snowex_db/point_data.py

```diff
@@ -37,8 +36,15 @@ def __init__(
         """
         self._row_based_timezone = row_based_timezone
         self._in_timezone = timezone
+        self._timezonefinder = None
         super().__init__(variable, meta_parser)
 
+    @property
+    def timezonefinder(self):
+        if self._timezonefinder is None:
+            self._timezonefinder = TimezoneFinder(in_memory=True)
+        return self._timezonefinder
+
```

Comment:
Creating one instance is expensive as the class loads file on disk. It does have the option to store that in memory.